app/gosqs/send_message_test.go

```diff
@@ -0,0 +1,219 @@
+package gosqs
+
+import (
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/Admiral-Piett/goaws/app"
+	"github.com/Admiral-Piett/goaws/app/fixtures"
+	"github.com/Admiral-Piett/goaws/app/interfaces"
+	"github.com/Admiral-Piett/goaws/app/models"
+	"github.com/Admiral-Piett/goaws/app/utils"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSendMessageV1_Success(t *testing.T) {
+	app.CurrentEnvironment = fixtures.LOCAL_ENVIRONMENT
+	defer func() {
+		utils.ResetApp()
+		utils.REQUEST_TRANSFORMER = utils.TransformRequest
+	}()
+
+	sendMessageRequest_success := models.SendMessageRequest{
+		QueueUrl:    "http://localhost:4200/new-queue-1",
+		MessageBody: "Test Message",
+	}
+	utils.REQUEST_TRANSFORMER = func(resultingStruct interfaces.AbstractRequestBody, req *http.Request) (success bool) {
+		v := resultingStruct.(*models.SendMessageRequest)
+		*v = sendMessageRequest_success
+		return true
+	}
+
+	q := &app.Queue{
+		Name:               "new-queue-1",
+		MaximumMessageSize: 1024,
+	}
+	app.SyncQueues.Queues["new-queue-1"] = q
+
+	_, r := utils.GenerateRequestInfo("POST", "/", nil, true)
+	status, response := SendMessageV1(r)
+
+	// Check the queue
+	assert.Equal(t, 1, len(q.Messages))
+	msg := q.Messages[0]
+	assert.Equal(t, "Test Message", string(msg.MessageBody))
+
+	// Check the response
+	assert.Equal(t, http.StatusOK, status)
+	sendMessageResponse, ok := response.(models.SendMessageResponse)
+	assert.True(t, ok)
+	assert.NotEmpty(t, sendMessageResponse.Result.MD5OfMessageBody)
+	// No FIFO Sequence
+	assert.Empty(t, sendMessageResponse.Result.SequenceNumber)
+}
+
+func TestSendMessageV1_Success_FIFOQueue(t *testing.T) {
+	app.CurrentEnvironment = fixtures.LOCAL_ENVIRONMENT
+	defer func() {
+		utils.ResetApp()
+		utils.REQUEST_TRANSFORMER = utils.TransformRequest
+	}()
+
+	sendMessageRequest_success := models.SendMessageRequest{
+		QueueUrl:    "http://localhost:4200/new-queue-1",
+		MessageBody: "Test Message",
+	}
+	utils.REQUEST_TRANSFORMER = func(resultingStruct interfaces.AbstractRequestBody, req *http.Request) (success bool) {
+		v := resultingStruct.(*models.SendMessageRequest)
+		*v = sendMessageRequest_success
+		return true
+	}
+
+	q := &app.Queue{
+		Name:               "new-queue-1",
+		MaximumMessageSize: 1024,
+		IsFIFO:             true,
+	}
+	app.SyncQueues.Queues["new-queue-1"] = q
+
+	_, r := utils.GenerateRequestInfo("POST", "/", nil, true)
+	status, response := SendMessageV1(r)
+
+	// Check the queue
+	assert.Equal(t, 1, len(q.Messages))
+	msg := q.Messages[0]
+	assert.Equal(t, "Test Message", string(msg.MessageBody))
+
+	// Check the response
+	assert.Equal(t, http.StatusOK, status)
+	sendMessageResponse, ok := response.(models.SendMessageResponse)
+	assert.True(t, ok)
+	assert.NotEmpty(t, sendMessageResponse.Result.MD5OfMessageBody)
+	// Should have FIFO Sequence
+	assert.NotEmpty(t, sendMessageResponse.Result.SequenceNumber)
+}
+
+func TestSendMessageV1_Success_Deduplication(t *testing.T) {
```

Comment:
I have added a test for Message Deduplication, but it looks goaws is not supported the latest AWS interface, right?

On current code:
Apply `EnableDuplicates` on the Queue

Latest AWS:
Apply `DeduplicationScope` on the Queue
https://docs.aws.amazon.com/ja_jp/AWSSimpleQueueService/latest/APIReference/API_CreateQueue.html#API_CreateQueue_RequestParameters

Reply:
Yeah good point - probably outdated.  I'll try to get to that at some point.  I only asked for the test because the code supports the skipping of the message if it's a duplicate, but you're right - in real life, it could use some work.